### PR TITLE
Allow pre- and post- configuration to be set

### LINKED ISSUE #44: Allow default configuration values in BeejuCore to be overridden
**Story**
As a user of BeeJU
I would like to be able to override the default configuration values
So that I can customise certain values to my own setup.

**Background**
Sometimes we use BeeJU alongside [HiveRunner](https://github.com/klarna/HiveRunner/) and need to use HiveRunner's `HiveConf` values from BeeJU so that they both refer to the same metastore database etc. BeeJU currently allows a custom `Map<String, String> configuration` to be passed in but any values set in this are overridden by BeeJU's own defaults. The user should be able to provide both "pre" and "post" configuration maps so that they can specify configuration they want BeeJU to override as well as configuration that they want to override BeeJU's own.

### PR DESCRIPTION
**Pull request by massdosage**
Fixes #44.

Also upgrades all non-rule tests to use JUnit5 instead of JUnit4.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Expedia Inc.
-Copyright (C) 2015-2017 Expedia Inc.
+Copyright (C) 2015-2020 Expedia, Inc.
 
-This product includes software developed by Expedia Inc. licensed under the Apache License, Version 2.0.
+This product includes software developed by Expedia, Inc. licensed under the Apache License, Version 2.0.
 
 This product includes software developed by The Apache Software Foundation (http://www.apache.org/) licensed under the Apache License, Version 2.0.
```
Comment (abhimanyugupta07) [inline NOTICE:6]:
```suggestion
This product includes software developed by The Apache Software Foundation (http://www.apache.org/) licensed under the Apache License, Version 2.0.

```

Comment (massdosage) [reply NOTICE:6]: I'm blind, what actually change here?

**Review by abhimanyugupta07 — APPROVED** [review]

File: NOTICE (hunk 1 above)
Comment (abhimanyugupta07) [reply NOTICE:6]: missing new line

Comment (abhimanyugupta07) [reply NOTICE:6]: oh, wrong suggestion.

Comment (massdosage) [reply NOTICE:6]: OMG, what? We can't allow that to happen! :)

Comment (abhimanyugupta07) [reply NOTICE]: Absolutely not !

File: src/test/java/com/hotels/beeju/core/BeejuCoreTest.java [hunk 2]
```diff
@@ -29,17 +29,49 @@
...
 import org.apache.hadoop.hive.metastore.api.Database;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
 
 public class BeejuCoreTest {
+  
+  private String preKey = "my.custom.pre.key"; 
+  private String preValue = "my.custom.pre.value";
+  private String postKey = "my.custom.pre.key"; 
+  private String postValue = "my.custom.pre.value";
```
Comment (JayGreeeen) [inline src/test/java/com/hotels/beeju/core/BeejuCoreTest.java:39]: should these say `my.custom.post.key` and `my.custom.post.value` ?

Comment (massdosage) [reply src/test/java/com/hotels/beeju/core/BeejuCoreTest.java:39]: Yes they should, well spotted. will update now.

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,6 @@ cache:
   directories:
     - ~/.m2/repository
 install:
-  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V -P travis
+  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true --quiet --batch-mode --show-version --activate-profiles coveralls
```
Comment (nvitucci) [inline .travis.yml:10]: Long options preferred?

Comment (massdosage) [reply .travis.yml:10]: Always ;) So we don't have to refer to the docs to remember what the hell `-B` means etc.

**Review by nvitucci — APPROVED** [review]

**Pull request merged**